src/test/java/org/apache/datasketches/quantilescommon/PartitionBoundariesTest.java

```diff
@@ -196,6 +196,40 @@ public void checkSimpleEndsAdjustment() {
     assertEquals(minItm, "1");
   }
 
+  @Test(expectedExceptions = SketchesArgumentException.class)
+  public void checkSketchPartitionLimits() {
+    final long totalN = 1_000_000;
+    final Comparator<String> comparator = Comparator.naturalOrder();
+    final ArrayOfStringsSerDe serDe = new ArrayOfStringsSerDe();
+    final KllItemsSketch<String> sk = KllItemsSketch.newHeapInstance(comparator, serDe);
+    final int d = digits(totalN);
+    for (int i = 1; i <= totalN; i++) {
+      sk.update(getString(i, d));
+    }
+    final int numLimit = sk.getMaxPartitions();
+    final int ret = sk.getNumRetained();
+    println("ret: " + ret + ", numLimit " + numLimit);
+    @SuppressWarnings("unused")
+    GenericPartitionBoundaries<String> gpb = sk.getPartitionBoundariesFromNumParts(numLimit + 1);
+  }
+
+  @Test(expectedExceptions = SketchesArgumentException.class)
+  public void checkSketchPartitionLimits2() {
+    final long totalN = 1_000_000;
+    final Comparator<String> comparator = Comparator.naturalOrder();
+    final ArrayOfStringsSerDe serDe = new ArrayOfStringsSerDe();
+    final KllItemsSketch<String> sk = KllItemsSketch.newHeapInstance(comparator, serDe);
+    final int d = digits(totalN);
+    for (int i = 1; i <= totalN; i++) {
+      sk.update(getString(i, d));
+    }
+    final long sizeLimit= sk.getMinPartitionSizeItems();
+
+    println("Min Size Limit: " + sizeLimit);
+    @SuppressWarnings("unused")
+    GenericPartitionBoundaries<String> gpb = sk.getPartitionBoundariesFromPartSize(sizeLimit - 1);
```

Comment:
do we ever test this method so that it succeeds rather than throwing?

Reply:
I tested it manually, and it passes at sizeLimit and fails as sizeLimit - 1. 
I didn't think it was necessary, but I could add a test that passes exactly at sizeLimit.  
Same thing goes for the limit on num partitions.

Reply:
Fixed.